ts/Series/ArcDiagram/ArcDiagramSeries.ts

```diff
@@ -0,0 +1,782 @@
+const {
+    extend,
+    merge,
+    pick,
+    relativeLength
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.arcdiagram
+ *
+ * @augments Highcharts.seriesTypes.sankey
+ */
+class ArcDiagramSeries extends SankeySeries {
+
+    /**
+     *  Arc diagram series is a chart drawing style in which
+     *  the vertices of the chart are positioned along a line
+     *  on the Euclidean plane and the edges are drawn as a semicircle
+     *  in one of the two half-planes delimited by the line,
+     *  or as smooth curves formed by sequences of semicircles.
+     *
+     * @sample highcharts/demo/arc-diagram/
+     *         Arc Diagram
+     *
+     * @extends      plotOptions.sankey
+     * @since        next
+     * @product      highcharts
+     * @requires     modules/arc-diagram
+     * @exclude      curveFactor, connectEnds, connectNulls, colorAxis, colorKey,
+     *               dataSorting, dragDrop, getExtremesFromAll, nodePadding,
+     *               centerInCategory, pointInterval, pointIntervalUnit,
+     *               pointPlacement, pointStart, relativeXValue, softThreshold,
+     *               stack, stacking, step, xAxis, yAxis
+     * @optionparent plotOptions.arcdiagram
+     */
+    public static defaultOptions: ArcDiagramSeriesOptions = merge(SankeySeries.defaultOptions, {
+
+        /**
+         * The option to center links rather than position them
+         * one after another
+         *
+         * @type    {boolean}
+         * @since next
+         * @default false
+         * @product highcharts
+         */
+        centeredLinks: false,
+
+        /**
+         * The radius of the link arc. If not set,
+         * it is calculated automatically.
+         *
+         * @type    {boolean}
+         * @since next
+         * @default undefined
+         * @product highcharts
+         * @apioption series.arcdiagram.linkRadius
+         */
+
+        /**
+         * The offset of an arc diagram nodes column in relation
+         * to the plotArea. The offset equal to 50% places nodes
+         * in the center of a chart.
+         *
+         * @type    {string}
+         * @since next
+         * @default undefined
+         * @product highcharts
+         * @apioption series.arcdiagram.offset
+         */
+
+        /**
+         * The global link weight. If not set, width will is calculated
+         * per link, depending on the weight value.
+         *
+         * @type    {number}
+         * @since next
+         * @default undefined
+         * @product highcharts
+         * @apioption series.arcdiagram.linkWeight
+         */
+
+        /**
+         * Whether nodes with different values should have the same size.
+         *
+         * @type    {boolean}
+         * @since next
+         * @default false
+         * @product highcharts
+         */
+        equalNodes: false,
+        keys: ['from', 'to', 'weight'],
```

Comment:
The `keys` should not be part of the default options. Use `pointArrayMap` instead. In this case I think it should be inherited.